cfpb/cfgov-refresh
# Allow more flexible date formats in Filterable List validation

This is part of the UI Improvements project, which is improving the user experience for filterable lists.

## Changes

- Allow single-digit month and day in filterable list date input. e.g. 10/2/2018, 1/31/2009
- Allow two-digit year in date inputs. e.g. 10/10/09

## Additions

- Add a few front-end tests to demonstrate which date formats are allowed

## Testing

1. With this branch running, visit http://localhost:8000/about-us/newsroom/
2. Try inputting different dates into the Date Rage "From:" and "To:" fields.
3. See that the following date formats all allow you to submit, and get a valid response:
     - 10/25/16 (two-digit year)
     - 9/1/16 (single-digit month and day)
     - 10/25/2016 (four-digit year)
     - 9/1/2016
     - 10/2016 (just month and year)
     - 7/2017
     - 10/16, 4/18 (month and two-digit year)
     - 2016


## Checklist


## Testing checklist

Should not change any visual behavior (except showing an error box under fewer conditions)
